src/client/lazy-app/Compress/Modal/index.tsx

```diff
@@ -0,0 +1,148 @@
+import { h, Component } from 'preact';
+import * as style from './style.css';
+import 'add-css:./style.css';
+import { linkRef } from 'shared/prerendered-app/util';
+import {
+  InfoIcon,
+  DiamondStarIcon,
+  ModalErrorIcon,
+} from 'client/lazy-app/icons';
+
+interface Props {}
+
+export interface ModalMessage {
+  type: 'info' | 'error' | 'update';
+  title: string;
+  content: string;
```

Comment:
I don’t think `content` should be a string, but rather a `VNode`. With string we lose the ability to have more complex interactivity in the modal (event listeners would get lost etc). Even `title` should probably be a `VNode`.

The next logical step is to make the Modal itself actually unaware of whether it’s an error or info or update. It’s just a component for modal dialogs, and since we can pass arbitrary VNodes for title and content, the user can style it depending on what the use-case is.

Reply:
Ahh you're right, `VNode` would be better suited. Will do!